### PR TITLE
Separate integration tests into authenticated and unauthenticated versions

### PR DESCRIPTION
**Pull request by synackd**
With the added feature of being able to enable JWT authentication in BSS and SMD, integration tests need to be added to test both unauthenticated and authenticated requests. This PR separates the hurl integration tests in `lanl/docker-compose/tests/all` into `lanl/docker-compose/tests/auth` and `lanl/docker-compose/tests/noauth` for authenticated and unauthenticated tests, respectively. The authenticated tests simply test that trying to access protected endpoints without a JWT returns a 401 Unauthorized.

This PR also:

- Changes the healthcheck commands to use long options to be clear about what is happening
- Changes the healthcheck command for SMD to use `curl` instead of `wget` (v2.14.2 adds `curl` to the container image)
- Adds `hostname` and `container_name` to services without it to have consistent container naming
- Updates instructions in `lanl/docker-compose/README.md`
- Updates the BSS/SMD image versions in the docker compose files
- Fixes incorrect delimiters for hydra in the `POSTGRES_MULTIPLE_DATABASES` env var in `ochami-services.yml`

### CODE REVIEW

**Comment by synackd** [comment]
Rebasing to include hydra healthcheck from #16.

**Comment by synackd** [comment]
Rebased again and added env vars for BSS OAuth2 integration, plus dependency on Hydra in ochami-services.yml.

File: lanl/docker-compose/README.md [hunk 1]
```diff
@@ -1,43 +1,114 @@
...
    ```
 
-   to bring up ochami services.
+   **NOTE:** Authenticated BSS and SMD run on ports 27778 and 27779,
+   respectively.
+
+   To run both authenticated and non-authenticated services simultaneously:
+
+   ```
+   docker compose -f ochami-services.yml -f ochami-krakend-ce.yml -f ochami-services-noauth.yml -f hydra.yml up
```
Comment (davidallendj) [inline lanl/docker-compose/README.md:39]: Does this run both auth/no-auth versions of SMD and BSS as well?

File: lanl/docker-compose/ochami-services.yml [hunk 2]
```diff
@@ -122,16 +123,21 @@ services:
       - BSS_DBPORT=5432
       - BSS_DBUSER=bss-user
       - BSS_DBPASS=${BSS_POSTGRES_PASSWORD} # Set in .env file
+      - BSS_AUTH_REQUIRED=true
```
Comment (davidallendj) [inline lanl/docker-compose/ochami-services.yml:126]: Did we remove this variable in favor of just using `BSS_JWKS_URL` to indicate that  auth was required?

**Review by davidallendj — COMMENTED** [review]
It might be worth having SMD initialize first to fetch the JWKS from Hydra so it can generate it's keys only once. Then, have other services depend on SMD (if they are making requests to SMD) and fetch their JWKS afterwards. That might solve the issue of multiple services trying to fetch the JWKS roughly at the same time.

File: lanl/docker-compose/README.md (hunk 1 above)
Comment (synackd) [reply lanl/docker-compose/README.md:39]: Yes. `ochami-services.yml` runs BSS/SMD with auth enabled, `ochami-services.yml` runs both with auth disabled.

File: lanl/docker-compose/ochami-services.yml (hunk 2 above)
Comment (synackd) [reply lanl/docker-compose/ochami-services.yml:126]: Yep. Setting `BSS_JWKS_URL` or passing `--jwks-url` will enable auth.

File: lanl/docker-compose/README.md (hunk 1 above)
Comment (davidallendj) [reply lanl/docker-compose/README.md:39]: I'm assuming you meant `ochami-services-no-auth.yml` runs both with auth disabled.

Comment (synackd) [reply lanl/docker-compose/README.md:39]: Whoops, yes. Corrected.

File: lanl/docker-compose/ochami-services.yml (hunk 2 above)
Comment (synackd) [reply lanl/docker-compose/ochami-services.yml]: Removed in  cc5ae5b.

**Comment by synackd** [comment]
Rebased to incorporate #28 for hydra.

**Comment by synackd** [comment]
Rebased and incorporated SMD changes from v1.15.0 (https://github.com/OpenCHAMI/smd/pull/17)

**Comment by synackd** [comment]
Cleaned up some commits and rebased.

**Review by travisbcotton — APPROVED** [review]
Manually ran nightly tests and they passed. Looks good to merge to me.

**Comment by synackd** [comment]
LGTM then!

**Pull request merged**